e2e/test/scenarios/dashboard-filters/dashboard-filters-sql-number.cy.spec.js

```diff
@@ -90,3 +90,69 @@ describe("scenarios > dashboard > filters > SQL > text/category", () => {
     });
   });
 });
+
+describe("scenarios > dashboard > filters > SQL > number", () => {
+  const questionDetails = {
+    name: "Question 1",
+    native: {
+      query:
+        "SELECT * from products where true [[ and price > {{price}}]] [[ and rating > {{rating}} ]] limit 5;",
+      "template-tags": {
+        price: {
+          type: "number",
+          name: "price",
+          id: "b22a5ce2-fe1d-44e3-8df4-f8951f7921bc",
+          "display-name": "Price",
+        },
+        rating: {
+          type: "number",
+          name: "rating",
+          id: "b22a5ce4-fe1d-44e3-8df4-f8951f7921bc",
+          "display-name": "Rating",
+        },
+      },
+    },
+  };
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createNativeQuestionAndDashboard({ questionDetails }).then(
+      ({ body: { card_id, dashboard_id } }) => {
+        visitQuestion(card_id);
+
+        visitDashboard(dashboard_id);
+      },
+    );
+
+    editDashboard();
+  });
+
+  it("should keep filter value on blur (metabase#31975)", () => {
+    setupNumberFilter("Price");
+    setupNumberFilter("Rating");
+
+    saveDashboard();
+
+    cy.findByPlaceholderText("Price").type("95").blur();
+    cy.findByPlaceholderText("Rating").type("3.8").blur();
+
+    cy.findAllByTestId("table-row")
+      .should("have.length", 2)
+      .and("contain", "Doohickey")
+      .and("contain", "Widget");
```

Comment:
nit: It might not look apparent from reading the test alone without running it why `Doohikey` and `Widget` would be asserted here. What do you think about asserting the price and rating themselves? e.g. price = 98.82 and 95.93 and rating = 4.3 and 4.4. This way it'd be easier to understand that the filter works as expected.